translator/metrics/otlp_to_oc.go

```diff
@@ -0,0 +1,292 @@
+}
+
+func metricToOC(metric *otlpmetrics.Metric) *ocmetrics.Metric {
+	return &ocmetrics.Metric{
+		MetricDescriptor: descriptorToOC(metric.MetricDescriptor),
+		Timeseries:       dataPointsToTimeseries(metric),
+		Resource:         nil,
+	}
+}
+
+func descriptorToOC(descriptor *otlpmetrics.MetricDescriptor) *ocmetrics.MetricDescriptor {
+	if descriptor == nil {
+		return nil
+	}
+
+	return &ocmetrics.MetricDescriptor{
+		Name:        descriptor.Name,
+		Description: descriptor.Description,
+		Unit:        descriptor.Unit,
+		Type:        descriptorTypeToOC(descriptor.Type),
+	}
+}
+
+func descriptorTypeToOC(t otlpmetrics.MetricDescriptor_Type) ocmetrics.MetricDescriptor_Type {
+	switch t {
+	case otlpmetrics.MetricDescriptor_UNSPECIFIED:
+		return ocmetrics.MetricDescriptor_UNSPECIFIED
+	case otlpmetrics.MetricDescriptor_GAUGE_INT64:
+		return ocmetrics.MetricDescriptor_GAUGE_INT64
+	case otlpmetrics.MetricDescriptor_GAUGE_DOUBLE:
+		return ocmetrics.MetricDescriptor_GAUGE_DOUBLE
+	case otlpmetrics.MetricDescriptor_GAUGE_HISTOGRAM:
+		return ocmetrics.MetricDescriptor_GAUGE_DISTRIBUTION
+	case otlpmetrics.MetricDescriptor_COUNTER_INT64:
+		return ocmetrics.MetricDescriptor_CUMULATIVE_INT64
+	case otlpmetrics.MetricDescriptor_COUNTER_DOUBLE:
+		return ocmetrics.MetricDescriptor_CUMULATIVE_DOUBLE
+	case otlpmetrics.MetricDescriptor_CUMULATIVE_HISTOGRAM:
+		return ocmetrics.MetricDescriptor_CUMULATIVE_DISTRIBUTION
+	case otlpmetrics.MetricDescriptor_SUMMARY:
+		return ocmetrics.MetricDescriptor_SUMMARY
+	default:
+		return invalidMetricDescriptorType
+	}
+}
+
+func dataPointsToTimeseries(metric *otlpmetrics.Metric) []*ocmetrics.TimeSeries {
+	length := len(metric.Int64Datapoints) + len(metric.DoubleDatapoints) + len(metric.HistogramDatapoints) +
+		len(metric.SummaryDatapoints)
+	if length == 0 {
+		return nil
+	}
+
+	timeseries := make([]*ocmetrics.TimeSeries, 0, length)
+	for _, point := range metric.Int64Datapoints {
+		ts := int64PointToOC(point)
+		timeseries = append(timeseries, ts)
+	}
+	for _, point := range metric.DoubleDatapoints {
+		ts := doublePointToOC(point)
+		timeseries = append(timeseries, ts)
+	}
+	for _, point := range metric.HistogramDatapoints {
+		ts := histogramPointToOC(point)
+		timeseries = append(timeseries, ts)
+	}
+	for _, point := range metric.SummaryDatapoints {
+		ts := summaryPointToOC(point)
+		timeseries = append(timeseries, ts)
+	}
+
+	return timeseries
+}
+
+func summaryPointToOC(point *otlpmetrics.SummaryDataPoint) *ocmetrics.TimeSeries {
+	return &ocmetrics.TimeSeries{
+		StartTimestamp: translatorcommon.UnixnanoToTimestamp(point.Value.StartTimeUnixnano),
+		LabelValues:    labelsToOC(point.LabelValues),
+		Points: []*ocmetrics.Point{
+			{
+				Timestamp: translatorcommon.UnixnanoToTimestamp(point.Value.TimestampUnixnano),
+				Value: &ocmetrics.Point_SummaryValue{
+					SummaryValue: &ocmetrics.SummaryValue{
+						Count: int64Value(point.Value.Count),
+						Sum:   doubleValue(point.Value.Sum),
```

Comment:
This changed with https://github.com/open-telemetry/opentelemetry-collector/pull/603. Will be point.Count and point.Sum